b2share/modules/records/triggers.py

```diff
@@ -60,12 +70,18 @@ def set_record_schema(record, **kwargs):
     record['$schema'] = community_schema_json_schema_link(schema)
 
 
+def set_record_owner(record, **kwargs):
+    if not current_user.is_authenticated:
```

Comment:
I think you are right for the `$schema` part. Regarding the bucket I don't know. I have to check if there is any issue with the loader creating other resources such as buckets.